pkg/llmstrategy/chatbased/strategy.go

```diff
@@ -39,84 +41,116 @@ type Strategy struct {
 
 	// PromptTemplateFile allows specifying a custom template file
 	PromptTemplateFile string
-	PreviousQueries    []string
 
 	// Recorder captures events for diagnostics
 	Recorder journal.Recorder
 
 	RemoveWorkDir bool
 
-	MaxIterations    int
-	CurrentIteration int
+	MaxIterations int
 
 	Kubeconfig          string
 	AsksForConfirmation bool
 
 	Tools tools.Tools
 }
 
-// ExecuteChatBased executes a chat-based agentic loop with the LLM using function calling.
-func (a *Strategy) RunOnce(ctx context.Context, query string, previousQueries []string, u ui.UI) error {
+type Conversation struct {
+	strategy *Strategy
+
+	// Recorder captures events for diagnostics
+	recorder journal.Recorder
+
+	UI      ui.UI
+	llmChat gollm.Chat
+
+	workDir string
+}
+
+func (s *Strategy) NewConversation(ctx context.Context, u ui.UI) (llmstrategy.Conversation, error) {
```

Comment:
I am bit confused about the hierarchy here. conversation has strategy and then strategy has a method to start a conversation .....

Reply:
Yeah, it could certainly use some improvement.  Conversation is a stateful discussion thread.  Strategy is the pluggable way we interact with the conversation.  We could have two conversations going on both with the same react strategy.

Reply:
Oh I see what you mean.  Yes, it turns out we need some variables from the "parent strategy".  Ideally we would not.  I managed to extract that in our eval framework by copying the variables.  We could see if that works here.  And maybe `parent` is a better name.